src/tree.h

```diff
@@ -17,15 +17,15 @@
 struct git_tree_entry {
 	uint16_t attr;
 	uint16_t filename_len;
-	git_oid oid;
-	bool pooled;
-	char filename[GIT_FLEX_ARRAY];
+	const git_oid *oid;
+	const char *filename;
 };
 
 struct git_tree {
 	git_object object;
+	git_odb_object *odb_obj;
+	git_array_t(git_tree_entry) entries_arr;
 	git_vector entries;
```

Comment:
We're keeping it so our homing search works. It expects to deal with a `git_vector` and I didn't want to deal with reimplementing the binary sorting algo on top of a different data structure.